src/cwl_platform/omics_platform.py

```diff
@@ -0,0 +1,358 @@
+        :param filters: Dictionary containing filter criteria
+            {
+                'name': 'file_name',
+                'prefix': 'file_prefix',
+                'suffix': 'file_suffix',
+                'folder': 'folder_name',
+                'recursive': True/False
+            }
+        :return: List of tuples (file path, file object) matching filter criteria
+        """
+        ''' TODO '''
+        return
+
+    def get_folder_id(self, project, folder_path):
+        '''
+        There is not unique ID for a folder in s3, so just return the folder_path
+        The one caveat is that Omics wants trailing slashes on folder paths, so add one.
+        '''
+        if folder_path.endswith("/"):
+            return folder_path
+        return folder_path + "/"
+
+    def get_task_input(self, task, input_name):
+        ''' Retrieve the input field of the task '''
+        self.logger.info("TBD: Getting input for task %s", task)
+        return None
+
+    def get_task_outputs(self, task):
+        self.logger.info('TBD: Getting task outputs %s', task)
+        return []
+
+    def get_task_state(self, task, refresh=False):
+        ''' 
+        Get status of run by task_id.
+        task: A dictionary of omics response from start_run. Includes Run ID, Name, Tags, etc.
+        return status of the run (Complete, Failed, Running, Cancelled, Queued).
+        '''
+
+        try:
+            run_info = self.api.get_run(id=task['id'])
+            job_status = run_info['status']
+        except:
+            raise ValueError('No Status information found for job %s. Check job status.', task['id'])
+
+        if job_status == 'COMPLETED':
+            return 'Complete'
+        if job_status == 'FAILED':
+            return 'Failed'
+        if job_status in ['STARTING','RUNNING','STOPPING']:
+            return 'Running'
+        if job_status in ['CANCELLED','DELETED']:
+            return 'Cancelled'
+        if job_status == 'PENDING':
+            return 'Queued'
+
+        raise ValueError('Unknown task state: %s : %s', task['id'], job_status)
+
+    def get_task_output(self, task, output_name):
+        ''' Retrieve the output field of the task '''
+        # Get output file mapping from outputs.json
+        taskinfo = self.api.get_run(id=task["id"])
+        output_json = taskinfo['outputUri'].split(self.output_bucket+'/')[1] + task["id"] + "/logs/outputs.json"
+        response = self.s3_client.get_object(Bucket=self.output_bucket, Key=output_json)
+        content = response['Body'].read().decode("utf-8")
+        mapping = json.loads(content)
+
+        if output_name not in mapping:
+            raise KeyError(f"Output field '{output_name}' not found in mapping file.")
+        all_outputs = mapping[output_name]
+        if isinstance(all_outputs, list):
+            outputs = [c["location"] for c in all_outputs]
+            return outputs
+        else:
+            return all_outputs["location"]
```

Comment:
Check for the existence of 'location' as a key before using or else this could result in a KeyError